python/ray/tests/test_runtime_env.py

```diff
@@ -286,6 +287,76 @@ def f():
         ray.get(a.ready.remote())
 
 
+def test_runtime_env_rescheduling(shutdown_only):
```

Comment:
Can you make sure all these scenarios are tested? (please verify you did it!)

- Regular rescheduling (failed once and make sure the task is scheduled on other node).
- Rescheduling failed because every node is blacklisted -> should raise exception
- Rescheduling failed because the only node that has available resources is blacklisted -> should raise exception